migrations/1614180464492-AddV4SignedOrdersAndPesistentSignedOrders.ts

```diff
@@ -0,0 +1,111 @@
+        );
+        await queryRunner.query(`ALTER TABLE "persistent_signed_orders" ALTER COLUMN "state" DROP DEFAULT`);
+        await queryRunner.query(
+            `ALTER TABLE "persistent_signed_orders" ALTER COLUMN "state" TYPE "persistent_signed_orders_state_enum_old" USING "state"::"text"::"persistent_signed_orders_state_enum_old"`,
+        );
+        await queryRunner.query(`ALTER TABLE "persistent_signed_orders" ALTER COLUMN "state" SET DEFAULT 'ADDED'`);
+        await queryRunner.query(`DROP TYPE "persistent_signed_orders_state_enum"`);
+        await queryRunner.query(
+            `ALTER TYPE "persistent_signed_orders_state_enum_old" RENAME TO  "persistent_signed_orders_state_enum"`,
+        );
+        await queryRunner.query(`ALTER TABLE "persistent_signed_orders" ALTER COLUMN "created_at" SET DEFAULT now()`);
+        await queryRunner.query(`COMMENT ON COLUMN "persistent_signed_orders"."created_at" IS NULL`);
+        await queryRunner.query(`ALTER TABLE "signed_orders" ALTER COLUMN "created_at" SET DEFAULT now()`);
+        await queryRunner.query(`COMMENT ON COLUMN "signed_orders"."created_at" IS NULL`);
+        await queryRunner.query(`DROP INDEX "IDX_7648bdf4a96ddc5cc7cfbc0135"`);
+        await queryRunner.query(`DROP INDEX "IDX_d611fdbc7da0effe48c103ac77"`);
+        await queryRunner.query(`DROP INDEX "IDX_3bd8111c0502847f405082b582"`);
+        await queryRunner.query(`DROP TABLE "persistent_signed_orders_v4"`);
+        await queryRunner.query(`DROP TYPE "persistent_signed_orders_v4_state_enum"`);
+        await queryRunner.query(`DROP TABLE "signed_orders_v4"`);
+        await queryRunner.query(
+            `CREATE INDEX "persistent_signed_orders_maker_address" ON "persistent_signed_orders" ("maker_address") `,
+        );
+        await queryRunner.query(
+            `CREATE INDEX "persistent_signed_orders_maker_asset_data" ON "persistent_signed_orders" ("maker_asset_data") `,
+        );
+        await queryRunner.query(
+            `CREATE INDEX "persistent_signed_orders_taker_asset_data" ON "persistent_signed_orders" ("taker_asset_data") `,
+        );
+        await queryRunner.query(
+            `CREATE INDEX "persistent_signed_orders_fee_recipient_address" ON "persistent_signed_orders" ("fee_recipient_address") `,
+        );
+        await queryRunner.query(`CREATE INDEX "maker_address_idx" ON "signed_orders" ("maker_address") `);
+        await queryRunner.query(`CREATE INDEX "taker_asset_data_idx" ON "signed_orders" ("taker_asset_data") `);
+        await queryRunner.query(`CREATE INDEX "maker_asset_data_idx" ON "signed_orders" ("maker_asset_data") `);
+        await queryRunner.query(
+            `CREATE INDEX "maker_taker_asset_data_idx" ON "signed_orders" ("maker_asset_data", "taker_asset_data") `,
+        );
+    }
+}
```

Comment:
**TODO:** Compare with prod to verify that we didn't lose any indexes with this migration